### DISCUSSION TITLE
On Datatable CRUD, how to get My Main Delete button's row Message to Update After an Edit?

### DISCUSSION THREAD

**Discussion by NCIDevRonR** [primefaces/community#213] [PrimeFaces] [answered]
Please reference this [reproducer](https://github.com/NCIDevRonR/primefaces-test/tree/cannot-select-row-on-data-table).
I've recently incorporated an observer pattern fix to enable and disable the main "Edit" and "Delete" buttons based on whether or not a row is selected.
After implementing this fix, playing user on this screen, if I select "Sand Point, Alaska", click the main Edit button, edit the city name to "Sandy Point", click "Save", then click the main Delete button, I get a prompt asking if I want to "Delete Sand Point, Alaska", where the name shown doesn't match the just-edited city name.  I notice if I click away then I click back, I see the updated city name, but how do I make sure the main Delete button's confirm message **_always_** displays the up-to-date version of the city name?

**Comment by FlipWarthog** [comment, marked as answer]
The issue is the confirm text isn't being updated after the edit. Continuing with the same observer pattern, your delete button (and its confirm) need to know that an edit happened. `p:autoUpdate` lets you add multiple events that you care about, so your delete can change to:

```xml
<p:autoUpdate on="selection-changed,selection-edited"/>
```

Then, the save button of your edit dialog can be:
```xml
<p:commandButton value="Save" icon="pi pi-check" actionListener="#{cityInfoController.save}"
                 update="@obs(selection-edited) manage-cityInfo" process="manage-cityInfo @this" />
```

Try this and see if it resolves the stale confirm issue.

**Reply by NCIDevRonR** [reply]
That fixed it!  Thanks for your help again.  I'm going to study this observer pattern stuff a little more closely, so it'll feel a little more intuitive.

**Reply by FlipWarthog** [reply]
Yeah, it's really nice once you get your head around it. So rather than having a button with `update"field1 panel2 otherSection3"` it simply updates an `@obs(event-name)` and all the other components that "care" about that action can subscribe. Basically inverting the dependency and cleaning things up.
